shared/rSharp.cpp

```diff
@@ -518,36 +520,62 @@ SEXP r_create_clr_object(SEXP p) {
 	get_FullTypeName(p, &ns_qualified_typename); p = CDR(p);
 	methodParams = p;
 
+	RSharpGenericValue** params = sexp_to_parameters(methodParams);
+	R_len_t numberOfObjects = Rf_length(methodParams);
+
+	try
+	{
+		auto return_value = createInstance(ns_qualified_typename, params, numberOfObjects);
+		free(ns_qualified_typename);
+		free_params_array(params, numberOfObjects);
+		return ConvertToSEXP(return_value);
+	}
+	catch (const std::exception& ex) 
+	{
+		free(ns_qualified_typename);
+		free_params_array(params, numberOfObjects);
+		error_return(ex.what())
+	}
+}
+
+RSharpGenericValue createInstance(char* ns_qualified_typename, RSharpGenericValue** parameters, const R_len_t numberOfObjects)
+{
+	RSharpGenericValue returnValue;
 
-	//if the function pointer has not been initialized, initialize it
 	if (create_instance_fn_ptr == nullptr)
 		initializeCreateInstance();
+
 	auto create_instance = reinterpret_cast<CreateInstanceDelegate>(create_instance_fn_ptr);
 
-	RSharpGenericValue** params = sexp_to_parameters(methodParams);
-	R_len_t num_objects = Rf_length(methodParams);
+	auto result = create_instance(ns_qualified_typename, parameters, numberOfObjects, &returnValue);
 
-	create_instance(ns_qualified_typename, params, num_objects, &return_value);
+	if (result < 0)
+	{
+		char* errorMessage = (char*)returnValue.value;
+		throw std::runtime_error(errorMessage);
+	}
 
-	free_params_array(params, num_objects);
-	return ConvertToSEXP(return_value);
+	return returnValue;
 }
 
-RSharpGenericValue callInstance(RSharpGenericValue** instance, const char* mnam, char* ns_qualified_typename, RSharpGenericValue** params, const R_len_t numberOfObjects)
+RSharpGenericValue callInstance(RSharpGenericValue** instance, const char* calledMethodName, char* ns_qualified_typename, RSharpGenericValue** parameters, const R_len_t numberOfObjects)
```

Comment:
```calledMethodName``` is much, much better than mnam. If you did not already replace all mnam with this, please do it! 👍

Reply:
Likewise ```parameters``` >> ```params```